pkg/driver/executor/executor.go

```diff
@@ -257,7 +275,73 @@ func (ex *Executor) deployServer(machineName string, userData []byte, nws []serv
 	return ex.Compute.CreateServer(createOpts)
 }
 
-func resourceInstanceBlockDevicesV2(rootDiskSize int, imageID string) ([]bootfromvolume.BlockDevice, error) {
+func (ex *Executor) instanceBlockDeviceWithType(name, imageID string) ([]bootfromvolume.BlockDevice, error) {
+	volume, err := ex.Storage.GetVolume(name)
+	if err != nil && !client.IsNotFoundError(err) {
+		return nil, err
+	}
+
+	if client.IsNotFoundError(err) {
+		volume, err = ex.Storage.CreateVolume(volumes.CreateOpts{
+			Name:             name,
+			VolumeType:       *ex.Config.Spec.RootDiskType,
+			Size:             ex.Config.Spec.RootDiskSize,
+			ImageID:          imageID,
+			AvailabilityZone: ex.Config.Spec.AvailabilityZone,
+			Metadata:         ex.Config.Spec.Tags,
+		})
+		if err != nil {
+			return nil, fmt.Errorf("failed to created volume [Name=%s]: %v", name, err)
+		}
+	}
+
+	err = ex.volumeWaitForStatus(volume.ID, []string{client.VolumeStatusCreating}, []string{client.VolumeStatusAvailable}, 600)
+	if err != nil {
+		return nil, err
+	}
+
+	blockDeviceOpts := make([]bootfromvolume.BlockDevice, 1)
+	blockDeviceOpts[0] = bootfromvolume.BlockDevice{
+		UUID:                volume.ID,
+		VolumeSize:          ex.Config.Spec.RootDiskSize,
+		BootIndex:           0,
+		DeleteOnTermination: true,
+		SourceType:          "volume",
+		DestinationType:     "volume",
+	}
+
+	return blockDeviceOpts, nil
+}
+
+func (ex *Executor) volumeWaitForStatus(volumeID string, pending, target []string, secs int) error {
+	return wait.Poll(time.Second, time.Duration(secs)*time.Second, func() (done bool, err error) {
+		current, err := ex.Storage.GetVolume(volumeID)
+		if err != nil {
+			if client.IsNotFoundError(err) && strSliceContains(target, client.ServerStatusDeleted) {
```

Comment:
```suggestion
			if client.IsNotFoundError(err) && strSliceContains(target, client.VolumeStatusDeleting) {
```

Reply:
This is a nice catch about the server status I used there, but I am not sure if using the deleting status and if that's enough to warranty that the volume is guaranteed to be deleted. I think the best way is to continue polling on deleting status until the volume is no longer found